db_whitebox_test.go

```diff
@@ -112,6 +119,180 @@ func TestMethodPage(t *testing.T) {
+				defer close(ch)
+
+				go func() {
+					for cErr := range ch {
+						cErrs = append(cErrs, cErr)
+					}
+				}()
+
+				tx.check(chkConfig, ch)
+				return nil
+			})
+			require.NoError(t, vErr)
+			require.Equal(t, 1, len(cErrs))
+			require.ErrorContains(t, cErrs[0], fmt.Sprintf("leaf page(%d)", victimPageId))
+		})
+	}
+}
+
+// corruptLeafPage write an invalid leafPageElement into the victim page.
+func corruptLeafPage(t testing.TB, db *DB, isPanic bool) common.Pgid {
+	t.Helper()
+
+	victimPageId := findVictimPageId(t, db)
+
+	victimPage, victimBuf, err := guts_cli.ReadPage(db.Path(), uint64(victimPageId))
+	require.NoError(t, err)
+	require.NotNil(t, victimPage)
+	require.NotNil(t, victimBuf)
+	require.True(t, victimPage.IsLeafPage())
+	require.True(t, victimPage.Count() > 0)
+
+	// Dumping random bytes in victim page for corruption.
+	copy(victimBuf[16:], generateCorruptionBytes(t, isPanic))
+	// Write the corrupt page to db file.
+	err = guts_cli.WritePage(db.Path(), victimBuf)
+	require.NoError(t, err)
+
+	return victimPageId
+}
+
+// findVictimPageId finds all the leaf pages of a bucket and picks a random leaf page as a victim to be corrupted.
+func findVictimPageId(t testing.TB, db *DB) common.Pgid {
+	t.Helper()
+	// Read DB's RootPage.
+	rootPageId, _, err := guts_cli.GetRootPage(db.Path())
+	require.NoError(t, err)
+	rootPage, _, err := guts_cli.ReadPage(db.Path(), uint64(rootPageId))
+	require.NoError(t, err)
+	require.True(t, rootPage.IsLeafPage())
+	require.Equal(t, 1, len(rootPage.LeafPageElements()))
+	// Find Bucket's RootPage.
+	lpe := rootPage.LeafPageElement(uint16(0))
+	require.Equal(t, uint32(common.BranchPageFlag), lpe.Flags())
+	k := lpe.Key()
+	require.Equal(t, "testBucket", string(k))
+	bucketRootPageId := lpe.Bucket().RootPage()
+	// Read Bucket's RootPage.
+	bucketRootPage, _, err := guts_cli.ReadPage(db.Path(), uint64(bucketRootPageId))
+	require.NoError(t, err)
+	require.Equal(t, uint16(common.BranchPageFlag), bucketRootPage.Flags())
+	// Retrieve Bucket's PageIds
+	var bucketPageIds []common.Pgid
+	for _, bpe := range bucketRootPage.BranchPageElements() {
+		bucketPageIds = append(bucketPageIds, bpe.Pgid())
+	}
+
+	victimPageId := bucketPageIds[rand.Intn(len(bucketPageIds))]
+	return victimPageId
+}
+
+// generateSampleDataInBucket fill in sample data into given bucket to create the given
+// number of leafPages. To control the number of leafPages, sample data are generated in order.
+func generateSampleDataInBucket(t testing.TB, bk *Bucket, lPages int) {
+	t.Helper()
+	currentKey := 1
+	currentVal := 100
+	for i := 0; i < lPages; i++ {
+		currentSize := common.PageHeaderSize
+		for {
+			err := bk.Put(convertIntIntoBytes(t, currentKey), convertIntIntoBytes(t, currentVal))
+			require.NoError(t, err)
+			currentSize += 16 + 4 + 4
+			if currentSize >= 4096 {
+				break
+			}
+			currentKey++
+			currentVal++
+		}
+	}
+}
+
+// generateCorruptionBytes returns random bytes to corrupt a page.
```

Comment:
@ahrtr @fuweid please review the docs, maybe it is not according to best practices ?